rpaas/nginx.py

```diff
@@ -97,25 +120,41 @@ def wait_healthcheck(self, host, timeout=30, manage_healthcheck=True):
             healthcheck_path = self.nginx_healthcheck_app_path.lstrip('/')
             expected_response = self.nginx_app_expected_healthcheck
             port = self.nginx_app_port
-        timeout = datetime.timedelta(seconds=timeout)
-        while True:
-            try:
-                self._nginx_request(host, healthcheck_path, port=port, expected_response=expected_response)
-                break
-            except:
-                now = datetime.datetime.now()
-                if now > t0 + timeout:
-                    raise
-                time.sleep(1)
-
-    def _nginx_request(self, host, path, headers=None, port=None, expected_response=None):
-        if not port:
-            port = self.nginx_manage_port
-        url = "http://{}:{}/{}".format(host, port, path)
-        if headers:
-            rsp = requests.get(url, timeout=2, headers=headers)
+        self._nginx_request(host, healthcheck_path, port=port, expected_response=expected_response)
+
+    @retry_request
+    def add_session_ticket(self, host, data, timeout=30):
+        self._nginx_request(host, 'session_ticket', data=data, method='POST', secure=True,
+                            expected_response='ticket was succsessfully added')
+
+    def _nginx_request(self, host, path, headers=None, port=None,
+                       expected_response=None, secure=False, method='GET', data=None):
+        params = {}
+        if secure:
+            if not port:
+                port = self.nginx_manage_port_tls
+            protocol = 'https'
+            self._ca_cert_file()
+            params['verify'] = self.ca_path
         else:
-            rsp = requests.get(url, timeout=2)
+            if not port:
+                port = self.nginx_manage_port
+            protocol = 'http'
+        url = "{}://{}:{}/{}".format(protocol, host, port, path)
+        if method not in ['POST', 'PUT', 'GET']:
+            raise NginxError("Unsupported method {}".format(method))
+        if headers:
+            params['headers'] = headers
+        if data:
+            params['data'] = data
+        rsp = getattr(requests, method.lower())(url, timeout=2, **params)
```

Comment:
You can use [`requests.request(method, url, kwargs)`](http://docs.python-requests.org/en/master/api/#requests.request) here.